### DISCUSSION TITLE
How to use createStore subscribeWithSelector immer in Typescript

### DISCUSSION THREAD

**Discussion by rhydian0x** [pmndrs/zustand#2309] [Q&A]
it looks like normal
![image](https://github.com/pmndrs/zustand/assets/42164477/d9134452-f054-4f1d-b5b8-ad2cf2e29cda)

but this case has an error
![image](https://github.com/pmndrs/zustand/assets/42164477/e1a8c981-90a2-4d9e-ad82-928bc0d8d545)

**Comment by dai-shi** [comment]
We don't have the exact case, but `middlewareTypes.test.tsx` might help.
https://github.com/pmndrs/zustand/blob/27bffb1fba9de4c958f4a4b4d31c20e54e579fc0/tests/middlewareTypes.test.tsx#L467
